### DISCUSSION TITLE
Error after running 'yarpmotorgui'

### DISCUSSION THREAD

**Discussion by AndreyGalitsin** [robotology/community#555] [Q&A] [answered]
Hello, I have an error after running **yarpmotorgui**. No one robot's part is not available. 
I have the following errors in the window: **Error while opening device for part icub/face**
Also in terminal I have the following output:

[ERROR] |yarp.os.YarpPluginSettings| Cannot find "remote_controlboard" plugin (not built in, and no .ini file found for it)Check that YARP_DATA_DIRS leads to at least one directory with plugins/remote_controlboard.ini or share/yarp/plugins/remote_controlboard.ini in it
[ERROR] |yarp.dev.PolyDriver| Could not find device <remote_controlboard>
[ERROR] Opening PolyDriver for part /icub/face failed...

Tell me please, how can I fix this problem?

**Comment by pattacini** [comment, marked as answer]
Hi @AndreyGalitsin 

It seems that your issue might be similar to #463.
Would you be willing to give a look at that and report any feedback?

**Reply by AndreyGalitsin** [reply]
Yes, I have seen this report. Unfortunately I don't know, how the software was stetted up into my laptop. yarpmotorgui application worked well last week, but now it doesn't, I can't understand why. 

I checked $YARP_DATA_DIRS:
```
icub@icubsrv:~$ echo $YARP_DATA_DIRS
/usr/local/src/robot/yarp/build/share/yarp:/usr/local/src/robot/icub-main/build/share/iCub:/usr/local/src/robot/iCubContrib/share/ICUBcontrib:/usr/local/src/robot/icub-tests/suits:/usr/share/yarp/plugins
```
File `remote_controlboard.ini` is located at the directory `/usr/share/yarp/plugins`, which is included into the YARP_DATA_DIRS
When I run yarpmotorgui, I have the following output:

![telegram-cloud-file-2-200413400084-274702-8844340690046186646](https://user-images.githubusercontent.com/55489582/135476322-afaed90b-1079-4569-8bad-2f52f9d01364.jpg)

Tell me please, what should I try to do to fix this problem?

**Reply by pattacini** [reply]
Did you try out to run:
```console
yarpdev --list | grep remote_controlboard
```
as suggested in that post?

What's the output?

Also, the content of your `YARP_DATA_DIRS` looks weird as it seems you have the components available from `/usr/local/src/robot/yarp/build`. You may try to remove the path `/usr/share/yarp/plugins` from the env var.

Perhaps, you issued a `make install` at some point, whereas you were using the build products instead.

**Reply by pattacini** [reply]
In this respect, you could also check the output of `which yarp`.

**Reply by AndreyGalitsin** [reply]
Hello! I came back to this problem. Now situation is the same. On the picture below, there are two outputs for your commands. Do You have any ideas what may be the reason of my problem?

![Screenshot from 2021-10-27 15-58-50](https://user-images.githubusercontent.com/55489582/139080995-c686967c-6bbc-49ed-a4dc-1edf18d520c1.png)

**Reply by pattacini** [reply]
Hi @AndreyGalitsin 

The first command crashed and that's clearly a symptom that your installation is not spot on.
Also, the version of `yarp` used is located in `/usr/bin`, whereas other YARP components are in `/usr/local`.

My suggestion is to clean up the system and carry out a full installation from scratch.
To this end, we recommend using the [`robotology-superbuild`](https://github.com/robotology/robotology-superbuild) way.

Here's the entry point in our documentation: https://icub-tech-iit.github.io/documentation/sw_installation.

**Comment by Nicogene** [comment]
Hi  @AndreyGalitsin,

when you run the `yarpmotorgui` you can select the parts you want to command. If you deselect `/icub/face`, it continues failing?

**Reply by pattacini** [reply]
Because of

>[ERROR] |yarp.dev.PolyDriver| Could not find device <remote_controlboard>

I think the problem might be upstream wrt the `yarpmotorgui` _per se_.

**Reply by Nicogene** [reply]
Ah yes sorry, I missed it, I thought was trying to open the `face` part and that robot didn't have it
